src/aggregator/aggregator/aggregator.go

```diff
@@ -151,6 +158,9 @@ func NewAggregator(opts Options) Aggregator {
 		metrics:           newAggregatorMetrics(scope, timerOpts, opts.MaxAllowedForwardingDelayFn()),
 		logger:            logger,
 	}
+
+	agg.tickShardFn = agg.tickShard
```

Comment:
nit: why not just set it in the struct initialization?

Reply:
Yeah maybe I'll refactor `tickShardFn` to take in the doneCh instead, then it doesn't have to be a method on the *aggregator